app/hooks/android_back_handler.tsx

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {EffectCallback, useEffect} from 'react';
+import {BackHandler} from 'react-native';
+
+import EphemeralStore from '@store/ephemeral_store';
+
+const useAndroidHardwareBackHandler = (componentId: string, callback: EffectCallback) => {
+    useEffect(() => {
+        const backHandler = BackHandler.addEventListener('hardwareBackPress', () => {
+            if (EphemeralStore.getNavigationTopComponentId() === componentId) {
+                callback();
+                return true;
+            }
+
+            return false;
+        });
+
+        return () => {
+            backHandler.remove();
+        };
+    }, [componentId]);
```

Comment:
I wonder if this should depend on the callback also. Depends on how it works on our use cases.

Reply:
Thank you.  I will keep this in mind when refactoring the other components.